engine/runtime/src/main/java/org/enso/interpreter/runtime/data/ArrayRope.java

```diff
@@ -106,6 +117,11 @@ public int size() {
       return cachedSize;
     }
 
+    @Override
+    public boolean isEmpty() {
+      return left.isEmpty() && right.isEmpty();
```

Comment:
```suggestion
      return size() == 0;
```

Why not re-use the logic for computing the size which includes caching?

The current implementation will need to traverse the whole rope on _every_ call, whereas with caching we can respond immediately on subsequent calls. And we already have this logic implemented.

Reply:
This logic is no longer needed, so removed. Also, it seemed that just finding the first non-empty segment was more efficient. But I agree that amortized cost would probably be better with calculating the size upfront.